### ISSUE TITLE
Problem with the sfml2 example

### ISSUE THREAD

**Issue by hachmeister** [mikke89/RmlUi#103]
I added a drawing command with a CircleShape to the sfml2 example. After that the window is only black if I start the sfml2 example.

See my changes:
https://github.com/mikke89/RmlUi/compare/master...hachmeister:sfml-drawing

**Comment by mikke89** [comment]
I played around with it for a bit. Looks like you need to (re)set the viewport before issuing SFML rendering commands. Eg. something along these lines:

```cpp
sf::View View;
View.setViewport(sf::FloatRect(0, 0, 1, 1));
MyWindow.setView(View);

MyWindow.draw(circle);
```

**Comment by hachmeister** [comment]
Thanks for your investigation with that I got it working.

But just some thoughts:
It would be nice not to have to adapt your program when using RmlUi.
In der RmlUiSFMLRenderer::Resize method a view is set. If I comment this out the example is still working, even with the circle. Is this view really necessary?

I made a second commit in my branch:
https://github.com/mikke89/RmlUi/compare/master...hachmeister:sfml-drawing

**Comment by mikke89** [comment]
Yeah, I'm not sure exactly why it is done like that. Perhaps it has something to do with the input handling? I don't want to just remove it until we know its implications. I'm sure there is room for improvements though, pull requests are very much welcome when you find something that works well ;)

**Comment by hachmeister** [comment]
I looked at the source code of SFML what happens when one sets a new view.

- SFML sets a flag in the RenderTarget class:
https://github.com/SFML/SFML/blob/master/src/SFML/Graphics/RenderTarget.cpp#L159

- if the flag is set the method applyCurrentView is called before the frame is drawn:
https://github.com/SFML/SFML/blob/master/src/SFML/Graphics/RenderTarget.cpp#L559

If the RenderInterface class from RmlUi and the SFML application sets a new view the last one will win.

If I set a view in a SFML application every frame this will also affect the RmlUi gui. Scales the view the SFML scene, then the RmlUi gui is also scaled. This is something I wouldn't expect.

I made a new branch in my RmlUi fork with the following changes:

The sfml example uses a view to display the scene and the window can be resized. It shows the scene with the same aspect ratio when you resize the window. You can use the keys 1, 2 and 3 to display the scene with different scale factors, 2 is the default factor.

The RenderInterface class set the Opengl viewport directly before the gui is drawn and not when the window is resized. So it does not interfere with the SFML view.

See my changes:
https://github.com/mikke89/RmlUi/compare/master...hachmeister:sfml-renderer

**Comment by mikke89** [comment]
Looks good! If you want to clean it up a bit, I'd like to see it as a pull request :)

**Comment by hachmeister** [comment]
I would make a new branch for the pull request. Should the changes to the example also be part of the pull request?

**Comment by mikke89** [comment]
Great. Yes, I like the little circles.

**Comment by mikke89** [comment]
Closing this as the issue was fixed in #106. Thanks!
